docs/advanced/document-store/milvus.md

```diff
@@ -0,0 +1,512 @@
+(milvus)=
+# Milvus
+
+One can use [Milvus](https://milvus.io/) as the Document store for DocumentArray. It is useful when one wants to have faster Document retrieval on embeddings, i.e. `.match()`, `.find()`.
+
+````{tip}
+This feature requires `pymilvus`. You can install it via `pip install "docarray[milvus]".` 
+````
+
+## Usage
+
+### Start Milvus service
+
+To use Milvus as the storage backend, you need a running Milvus server. You can use the following `docker-compose.yml`
+to start a Milvus server:
+
+`````{dropdown} docker-compose.yml
+
+```yaml
+version: '3.5'
+
+services:
+  etcd:
+    container_name: milvus-etcd
+    image: quay.io/coreos/etcd:v3.5.0
+    environment:
+      - ETCD_AUTO_COMPACTION_MODE=revision
+      - ETCD_AUTO_COMPACTION_RETENTION=1000
+      - ETCD_QUOTA_BACKEND_BYTES=4294967296
+      - ETCD_SNAPSHOT_COUNT=50000
+    volumes:
+      - ${DOCKER_VOLUME_DIRECTORY:-.}/volumes/etcd:/etcd
+    command: etcd -advertise-client-urls=http://127.0.0.1:2379 -listen-client-urls http://0.0.0.0:2379 --data-dir /etcd
+
+  minio:
+    container_name: milvus-minio
+    image: minio/minio:RELEASE.2022-03-17T06-34-49Z
+    environment:
+      MINIO_ACCESS_KEY: minioadmin
+      MINIO_SECRET_KEY: minioadmin
+    volumes:
+      - ${DOCKER_VOLUME_DIRECTORY:-.}/volumes/minio:/minio_data
+    command: minio server /minio_data
+    healthcheck:
+      test: ["CMD", "curl", "-f", "http://localhost:9000/minio/health/live"]
+      interval: 30s
+      timeout: 20s
+      retries: 3
+
+  standalone:
+    container_name: milvus-standalone
+    image: milvusdb/milvus:v2.1.4
+    command: ["milvus", "run", "standalone"]
+    environment:
+      ETCD_ENDPOINTS: etcd:2379
+      MINIO_ADDRESS: minio:9000
+    volumes:
+      - ${DOCKER_VOLUME_DIRECTORY:-.}/volumes/milvus:/var/lib/milvus
+    ports:
+      - "19530:19530"
+      - "9091:9091"
+    depends_on:
+      - "etcd"
+      - "minio"
+
+networks:
+  default:
+    name: milvus
+```
+
+`````
+
+Then
+
+```bash
+docker-compose up
+```
+
+You can find more installation guidance in the [Milvus documentation](https://milvus.io/docs/v2.1.x/install_standalone-docker.md).
+
+### Create DocumentArray with Milvus backend
+
+Assuming service is started using the default configuration (i.e. the server's gRPC address `http://localhost:19530`), you can 
+instantiate a DocumentArray with Milvus storage like so:
+
+```python
+from docarray import DocumentArray
+
+da = DocumentArray(storage='milvus', config={'n_dim': 10})
+```
+
+Here, `config` is configuration for the created Milvus collection,
```

Comment:
```suggestion
Here, `config` is configuration for the new Milvus collection,
```

Not essential, but I think it would be good to make clear that this is configuration for a newly-created collection, not a previously-created one.